src/lib/metricsMgr/MetricsManager.cpp

```diff
@@ -131,15 +134,54 @@ void MetricsManager::add(const std::string& srv, const std::string& subServ, con
     return;
   }
 
+  // FIXME P4: See github issue #2781
```

Comment:
In addition, I'd suggest to move these checking to an additional (class private?) function, so add() could be kept simple.